### ISSUE TITLE
Not using No Name buffer when using alpha/dashboard plugin

### ISSUE THREAD

**Issue by sQVe** [nvim-telescope/telescope.nvim#1474]
### Description

👋🏼 

If using either https://github.com/goolord/alpha-nvim or https://github.com/glepnir/dashboard-nvim and opening a file with *telescope* you still have a `[No Name]` buffer.

https://user-images.githubusercontent.com/2284724/142703001-e7d0e503-292c-4fbe-9474-78902713c73b.mp4

### Neovim version

```markdown
NVIM v0.6.0-dev+604-g725cbe7d41
Build type: RelWithDebInfo
LuaJIT 2.0.5
```

### Operating system and version

ArchLinux @ 5.15.2

### checkhealth telescope

```markdown
telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - OK: nvim-treesitter installed.

## Checking external dependencies
  - OK: rg: found ripgrep 13.0.0
  - OK: fd: found fd 8.2.1

## ===== Installed extensions =====

## Telescope Extension: `fzf`
  - INFO: No healthcheck provided
```

### Steps to reproduce

1. Install *telescope*.
2. Install either https://github.com/goolord/alpha-nvim or https://github.com/glepnir/dashboard-nvim.
3. Open a file via *telescope*.

### Expected behavior

Only one buffer should be opened just like if I open it with `:e`.

### Actual behavior

There's two buffers open, one for the selected file and one empty `[No Name]` buffer.

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      'goolord/alpha-nvim',
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')
  require('alpha').setup(require('alpha.themes.dashboard').opts)
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by Conni2461** [comment]
Just ran into the same with startify. Must be a recent regression. I will bisect this evening and see if i can find it.

Thanks for reporting :)

**Comment by Conni2461** [comment]
Commit that introduced this bug: b952c2f852de1ec5ce7306ccd9262c97e27902d2

Will fix it this evening

**Comment by Conni2461** [comment]
Fix should be in #1480
